### PR TITLE
[Hotfix][MySQL-CDC] Fix read gbk varchar chinese garbled characters

### LINKED ISSUE #7031: [Bug] [ZETA] debezium_json Chinese garbled characters
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### What happened

Hello! I am configuring the corresponding parameters in the instance, expecting to import 【MySQL CDC】 information into 【Kafka】, and the format used is compatible_debezium_json, but I found that there is garbled Chinese characters. How can I solve this?
I found that if I change the format in the sink to json, there are actually no Chinese garbled characters, but when both the source and sink are configured with the format as compatible_debezium_json, the problem arises.
like this:

expected =
        {"name": "常禄产权2",
        "registered_no": null,
        "short_name": "常禄产权2"}

actual = 
        {"name": "��»��Ȩ2",
        "registered_no": null,
        "short_name": "��»��Ȩ2"}

### SeaTunnel Version

2.3.5

### SeaTunnel Config

```conf
env {
  parallelism = 1
  job.mode = "STREAMING"
  checkpoint.interval = 10000
}

source {

  MySQL-CDC {
    base-url = "jdbc:mysql://10.0.0.136:3306/ihr_test?useUnicode=true&characterEncoding=UTF-8&useSSL=false&serverTimezone=GMT%2B8"
    hostname = 10.0.0.136
    username = "root"
    password = "123456"
    table-names = ["ihr_test.company"]
	
    catalog {
        factory=MySQL
    }

    startup.mode = "initial"

    # compatible_debezium_json options
    format = compatible_debezium_json
    debezium = {
        # include schema into kafka message
        key.converter.schemas.enable = false
        value.converter.schemas.enable = false
        # include ddl
        include.schema.changes = true
        # topic prefix
        database.server.name =  "keke_mysql_cdc"
    }
  }
}


sink {
  kafka {
      topic = "keke_topic_mysql_ihr_test"
      bootstrap.servers = "10.0.0.98:39092"
	  format = compatible_debezium_json
      semantics = AT_LEAST_ONCE
      kafka.config = {
        acks = "all"
        request.timeout.ms = 60000
      }
  }
}
```

### Running Command

```shell
seatunnel.cmd --config ./config/v2.batch.config.kafka.template -e local
```

### Error Exception

```log
There are no error messages in the console.
```

### Zeta or Flink or Spark Version

Zeta

### Java or Scala Version

java version "1.8.0_101"

### Screenshots

b706b6b1-45ad-4f16-ad93-a446519d9bdb

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by hailin0**
### Purpose of this pull request

[MySQL-CDC] Fix read gbk varchar chinese garbled characters

### Does this PR introduce _any_ user-facing change?
No

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-connectors-v2/connector-cdc/connector-cdc-mysql/src/main/java/org/apache/seatunnel/connectors/seatunnel/cdc/mysql/utils/MySqlUtils.java [hunk 1]
```diff
@@ -342,44 +334,6 @@ public static SeaTunnelRowType getSplitType(
         return getSplitType(primaryKeys.get(0), dbzConnectorConfig);
     }
 
-    /** Creates a new {@link MySqlDatabaseSchema} to monitor the latest MySql database schemas. */
-    public static MySqlDatabaseSchema createMySqlDatabaseSchema(
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-cdc/connector-cdc-mysql/src/main/java/org/apache/seatunnel/connectors/seatunnel/cdc/mysql/utils/MySqlUtils.java:346]: Remove duplicate methods

File: seatunnel-connectors-v2/connector-cdc/connector-cdc-mysql/src/main/java/org/apache/seatunnel/connectors/seatunnel/cdc/mysql/config/CustomMySqlConnectionConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+import java.sql.Connection;
+import java.sql.SQLException;
+
+import static io.debezium.connector.mysql.MySqlConnectorConfig.JDBC_DRIVER;
+
+public class CustomMySqlConnectionConfiguration
+        extends MySqlConnection.MySqlConnectionConfiguration {
+
+    protected static final String URL_PATTERN =
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-cdc/connector-cdc-mysql/src/main/java/org/apache/seatunnel/connectors/seatunnel/cdc/mysql/config/CustomMySqlConnectionConfiguration.java:33]: delete `characterSetResults=UTF-8` from

https://github.com/debezium/debezium/blob/393291bba54edc745f45a679ffd7b2aa7110f454/debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlConnection.java#L56

**Comment by hailin0** [comment]
image

**Review by Hisoka-X — APPROVED** [review]
LGTM

**Review by EricJoy2048 — APPROVED** [review]

**Pull request merged**